velox/functions/lib/aggregates/SetBaseAggregate.h

```diff
@@ -50,7 +50,15 @@ class SetBaseAggregate : public exec::Aggregate {
     for (auto i = 0; i < numGroups; ++i) {
       auto* group = groups[i];
       if (isNull(group)) {
-        arrayVector->setNull(i, true);
+        if constexpr (ignoreNulls) {
```

Comment:
I think we may use a new flag for this, as this flag is ture for CountDistinctAggregate.
https://github.com/facebookincubator/velox/blob/37aac8a96f9beede1622ffc3dbf4b45edca849e8/velox/functions/prestosql/aggregates/SetAggregates.cpp#L106
And please also need to update the spark function doc.

Reply:
I also noticed the usage of `CountDistinctAggregate` but am not sure about its semantics as it is an internal function used only for testing. I tested Presto count distinct and it returns `0` when no input, which aligns with current fix. @kagamiori Do you have any input? Thanks.

```
presto:test> select count(distinct col) FILTER (WHERE false) FROM (VALUES 1, 2, 1) AS tab(col);
 _col0 
-------
     0 
(1 row)
```

Reply:
Added a new flag 'nullForEmpty' to clarify the different behaviors between Presto and Spark.

CountDistinctAggregate overrides `extractValues` and handles empty input in it. Thanks.

https://github.com/facebookincubator/velox/blob/37aac8a96f9beede1622ffc3dbf4b45edca849e8/velox/functions/prestosql/aggregates/SetAggregates.cpp#L142-L144